### PR TITLE
Improves check for empty value for external db id in genes.

### PR DESCRIPTION
**Pull request by andricDu**

### CODE REVIEW

File: dcc-portal-ui/app/scripts/genes/views/gene.html [hunk 1]
```diff
@@ -143,7 +143,7 @@ <h1 data-ui-scrollfix=79><span class="t_badge t_badge__gene"></span>
                                 <i class="icon-external-link"></i> {{ id }}</a>
                         </li>
                     </ul>
-                    <span data-ng-if="!GeneCtrl.gene.externalDbIds['entrez_gene']">--</span>
+                    <span data-ng-if="!GeneCtrl.gene.externalDbIds['entrez_gene'] || GeneCtrl.gene.externalDbIds['entrez_gene'].length === 0">--</span>
```
Comment (btiernay) [inline dcc-portal-ui/app/scripts/genes/views/gene.html]: _.get (GeneCtrl, 'gene.externalDbIds.entrez_gene', []).length === 0

File: dcc-portal-ui/app/scripts/genes/views/gene.html [hunk 2]
```diff
@@ -155,7 +155,7 @@ <h1 data-ui-scrollfix=79><span class="t_badge t_badge__gene"></span>
                                 <i class="icon-external-link"></i> {{ id }}</a>
                         </li>
                     </ul>
-                    <span data-ng-if="!GeneCtrl.gene.externalDbIds['omim_gene']">--</span>
+                    <span data-ng-if="!GeneCtrl.gene.externalDbIds['omim_gene'] || GeneCtrl.gene.externalDbIds['omim_gene'].length === 0">--</span>
```
Comment (btiernay) [inline dcc-portal-ui/app/scripts/genes/views/gene.html]:
```javascript
_.get (GeneCtrl, 'gene.externalDbIds. omim_gene', []).length === 0
```

File: dcc-portal-ui/app/scripts/genes/views/gene.html [hunk 3]
```diff
@@ -167,7 +167,7 @@ <h1 data-ui-scrollfix=79><span class="t_badge t_badge__gene"></span>
                                 <i class="icon-external-link"></i> {{ id }}</a>
                         </li>
                     </ul>
-                    <span data-ng-if="!GeneCtrl.gene.externalDbIds['uniprotkb_swissprot']">--</span>
+                    <span data-ng-if="!GeneCtrl.gene.externalDbIds['uniprotkb_swissprot'] || GeneCtrl.gene.externalDbIds['uniprotkb_swissprot'].length === 0">--</span>
```
Comment (btiernay) [inline dcc-portal-ui/app/scripts/genes/views/gene.html]:
```javascript
_.get (GeneCtrl, 'gene.externalDbIds.uniprotkb_swissprot', []).length === 0
```

File: dcc-portal-ui/app/scripts/genes/views/gene.html (hunk 2 above)
Comment (btiernay) [inline dcc-portal-ui/app/scripts/genes/views/gene.html]: Btw, whats a `omim_gene`?

Comment (andricDu) [inline dcc-portal-ui/app/scripts/genes/views/gene.html]: http://omim.org/entry/180902

File: dcc-portal-ui/app/scripts/genes/js/genes.js [hunk 4]
```diff
@@ -52,13 +52,16 @@
...
     _ctrl.gvOptions = {location: false, panels: false, zoom: 50};
 
-
     _ctrl.gene = gene;
     _ctrl.gene.uiProteinTranscript = [];
     _ctrl.gene.fprojects = [];
     _ctrl.totalDonors = 0;
     _ctrl.gene.hasGVChromosome = GMService.isValidChromosome(_ctrl.gene.chromosome);
 
+    _ctrl.hasNoExternal = function(dbId) {
```
Comment (btiernay) [inline dcc-portal-ui/app/scripts/genes/js/genes.js]: Nice!

**Comment by btiernay** [comment]
```
app/scripts/genes/js/genes.js
  line 63  col 6  Missing semicolon.```

**Pull request merged**